File: admin/src/main/java/com/admin/service/QuizService.java
package com.admin.service;

import java.util.List;
import java.util.Set;

import com.admin.model.Category;
import com.admin.model.Quiz;

public interface QuizService {

	public Quiz addQuiz(Quiz quiz);
	
	public Quiz updateQuiz(Quiz quiz);
	
	public Set<Quiz> getQuizzes();
	
	public Quiz getQuiz(Long quizId);
	
	public void deleteQuiz(Long quizId);

	public List<Quiz> getQuizzesOfCategory(Category cat);
	
	public List<Quiz> getActiveQuizzes();
	
	public List<Quiz> getActiveQuizzesOfCategory(Category c);

	
}

File: frontend-ui/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { adminUrl } from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private _http: HttpClient) { }

  // get all quiz from server
  public quizzes() {
    return this._http.get(`${adminUrl}/quiz/`);
  }

    // get a quiz by its id
    public getQuiz(qId: any) {
      return this._http.get(`${adminUrl}/quiz/${qId}`);
    }

  // add quiz to server 
  public addQuiz(quiz: any) {
    console.log('from Quiz service');
    console.log(quiz);
    return this._http.post(`${adminUrl}/quiz/`, quiz);
  }

  // update a quiz
  public updateQuiz(quiz: any) {
    return this._http.put(`${adminUrl}/quiz/`, quiz);
  }

  // get quizzes of category
  public getQuizzesOfCategory(cid: any) {
    return this._http.get(`${adminUrl}/quiz/category/${cid}`);
  }

  // get active quizzes
  public getActiveQuizzes() {
    return this._http.get(`${adminUrl}/quiz/active`);
  }

  // get active quizzes of category
  public getActiveQuizzesOfCategory(cid: any) {
    return this._http.get(`${adminUrl}/quiz/category/active/${cid}`);
  }

  // delete a quiz
  public deleteQuiz(qid: any) {
    return this._http.delete(`${adminUrl}/quiz/${qid}`);
  }



}
